doc/pipeline-details.md

```diff
@@ -15,35 +15,45 @@ flowchart
+        tarball_node --> upload_tarball[Create and upload tarball to the storage <br />status=ready, result=None, set holdoff]
     end
     subgraph runner_service[Runner Service]
-        tarball_node --> |event: <br />tarball created <br />result=pass| runner[Runner]
-        runner --> test_node[Create 'test' node <br />status=pending, result=None]
+        upload_tarball --> |event: <br />tarball updated <br />status=ready| runner[Runner]
+        runner --> runner_node[Create build/test node <br />status=pending, result=None, holdoff=None]
     end
-    subgraph Run Tests
-        test_node --> runtime[Runtime Environment]
-        runtime --> update_test[Update 'test' node<br />status=completed, result=pass/fail]
+    subgraph Run Builds/Tests
+        runner_node --> runtime[Runtime Environment]
+        runtime --> set_ready[Update node<br />status=ready, result=None, set holdoff]
+        set_ready --> run_job[Run build/test job]
+        run_job --> job_done{Job done?}       
+        job_done --> |Yes| pass_runner_node[Update node<br />status=done, result=pass/fail/skip]
+        job_done --> |No| run_job
     end
-    subgraph trigger_completed_service[Trigger Completed Service]
-        update_test --> |event: <br />test node updated| trigger_completed[Trigger Completed]
-        trigger_completed --> node_completed{All child <br />nodes completed ?}
-        node_completed --> |Yes| generate_completed[Generate 'completed' <br />event]
+    subgraph set_completed_service[Set Completed Service]
+        get_pending_parent_nodes[Get nodes <br /> with status=ready]
+        set_completed[Set Completed] --> get_pending_parent_nodes
+        get_pending_parent_nodes --> hold_off_reached{Hold off reached?}
+        hold_off_reached --> |Yes| node_completed{All child <br />nodes completed ?}
+        node_completed --> |Yes| generate_completed[Set node <br />status=done, result=pass]
     end
-    tarball_node --> |event: <br />tarball created <br />status=completed| trigger_completed
     subgraph test_report_service[Test Report Service]
-        generate_completed --> |event: <br />completed| test_report[Test Report]
-        test_report --> email_report[Generate and <br />email test report]
+        test_report[Test Report] --> email_report[Generate and <br />email test report <br />for tarball]
     end
     subgraph set_timeout_service[Set Timeout Service]
-        set_timeout[Set Timeout] --> max_wait_time{Maximum wait <br/>time over?} --> |Yes| set_status[Set node status <br />to 'timeout']
-        max_wait_time --> |No| set_timeout[Set Timeout]
-        set_status --> |event: <br />updated| trigger_completed
+        set_timeout[Set Timeout] --> get_incomplete_nodes
+        get_incomplete_nodes[Get nodes <br /> with status=pending] --> max_wait_time
+        max_wait_time{Maximum wait <br/>time 'node.timeout' over?} --> |Yes| set_incomplete[Set parent and pending child nodes <br /> status=done, result=incomplete]
```

Comment:
I think `max_wait_time` is now just `timeout`, right?

Reply:
> I think `max_wait_time` is now just `timeout`, right?

Yes.